garage/tf/exploration_strategies/epsilon_greedy_strategy.py

```diff
@@ -0,0 +1,109 @@
+"""
+ϵ-greedy exploration strategy.
+
+Random exploration according to the value of epsilon.
+"""
+import numpy as np
+
+from garage.exploration_strategies import ExplorationStrategy
+from garage.misc.overrides import overrides
+from garage.spaces import Box
+
+
+class EpsilonGreedyStrategy(ExplorationStrategy):
+    """
+    ϵ-greedy exploration strategy.
+
+    Select action based on the value of ϵ. ϵ will decrease from
+    max_epsilon to min_epsilon within decay_ratio * total_step.
+
+    At state s, with probability
+    1 − ϵ: select action = argmax Q(s, a)
+    ϵ    : select a random action from an uniform distribution.
+
+    Args:
+        env_spec: Environment specification
+        total_step: Total steps in the training.
+        max_epsilon: The maximum(starting) value of epsilon.
+        min_epsilon: The minimum(terminal) value of epsilon.
+        decay_ratio: Fraction of total steps for epsilon decay.
+    """
+
+    def __init__(self,
+                 env_spec,
+                 total_step,
```

Comment:
Is `total_step` the same as `max_path_length`? If it is, please use `max_path_length` then.

Reply:
Got it. It'd be better to add this, i.e. max_path_length * n_epochs, in docstring I think.

Reply:
I can't seem to understand this equation, it looks related to your training architecture of dqn. Which is
```
for epoch in range(n_epochs):
    for t in range(n_training_steps):
```
but the regular way is:
```
for epoch in range(n_epochs):
    for epoch_cycle in range(n_epoch_cycles):
        for t in range(n_training_steps):
```
Please discuss the details about this because I think we need to have a way to circumvent this.

Reply:
After discussing with you, I think we can rename it to total_timesteps or include it is the same as total time steps in docstring.